test/sample-test/run_test.sh

```diff
@@ -101,242 +103,217 @@ while [ "$1" != "" ]; do
+  cd ${WORK_DIR}
+}
 
+################################################################################
+# Utility function to generate formatted result, and move it to a place for PROW
+# to pick up.
+# Arguments:
+#   Test name.
+################################################################################
+check_result() {
   cd "${TEST_DIR}"
-  python3 run_kubeflow_test.py --input ${BASE_DIR}/samples/core/kubeflow-tf/kubeflow-training-classification.zip --result $SAMPLE_KUBEFLOW_TEST_RESULT --output $SAMPLE_KUBEFLOW_TEST_OUTPUT --namespace ${NAMESPACE}
+  python3 run_sample_test.py --input ${WORK_DIR}/$1.yaml \
+  --result ${SAMPLE_TEST_RESULT} --output ${SAMPLE_TEST_OUTPUT} --testname $1 \
+  --namespace ${NAMESPACE}
 
   echo "Copy the test results to GCS ${RESULTS_GCS_DIR}/"
-  gsutil cp ${SAMPLE_KUBEFLOW_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_KUBEFLOW_TEST_RESULT}
-elif [ "$TEST_NAME" == "tfx" ]; then
-  SAMPLE_TFX_TEST_RESULT=junit_SampleTFXOutput.xml
-  SAMPLE_TFX_TEST_OUTPUT=${RESULTS_GCS_DIR}
-  
-  # Compile samples
-  cd ${BASE_DIR}/samples/core/tfx
-
-  dsl-compile --py taxi-cab-classification-pipeline.py --output taxi-cab-classification-pipeline.yaml
+  gsutil cp ${SAMPLE_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_TEST_OUTPUT}
+}
 
-  if [ -n "${DATAFLOW_TFT_IMAGE}" ];then
-    # Update the image tag in the yaml.
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tft:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_TFT_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tf-predict:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_PREDICT_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tfdv:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_TFDV_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tfma:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_TFMA_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-kubeflow-tf-trainer:\([a-zA-Z0-9_.-]\)\+|${KUBEFLOW_DNNTRAINER_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-kubeflow-deployer:\([a-zA-Z0-9_.-]\)\+|${KUBEFLOW_DEPLOYER_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:\([a-zA-Z0-9_.-]\)\+|${LOCAL_CONFUSIONMATRIX_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-roc:\([a-zA-Z0-9_.-]\)\+|${LOCAL_ROC_IMAGE}|g" taxi-cab-classification-pipeline.yaml
+################################################################################
+# Utility function to prepare and validate the results for sample test based on
+# ipynb.
+# Arguments:
+#   Test name.
+################################################################################
+check_notebook_result() {
+  jupyter nbconvert --to python $1.ipynb
+  pip3 install tensorflow==1.8.0
```

Comment:
This is not required by all notebook samples, maybe we can move it to the notebook sample file instead of in the test infra?

Reply:
Yes we discussed about it. I think it's better to do that in another PR since this one is already big enough. Do you agree?

Reply:
That's totally fine. How about add a TODO comment to keep track?

Reply:
Added TODO.